### DISCUSSION TITLE
Why is it so slow to create "createconnection" in a single process when it is called more than 20 times?. Version 5.21 does not have this problem.

### DISCUSSION THREAD

**Discussion by wj8400684** [rabbitmq/rabbitmq-dotnet-client#1230] [General]
var factory = new ConnectionFactory
{
    VirtualHost = ConnectionFactory.DefaultVHost,
    //AutomaticRecoveryEnabled = true,
    HostName = DefaultHost,
    UserName = DefaultUserName,
    Password = DefaultPassWord,
    Port = DefaultPort,
};

var watch = new Stopwatch();

for (int i = 0; i < 100; i++)
{
    watch.Restart();
    var rabbitmqConnection = factory.CreateConnection(Guid.NewGuid().ToString("n"));
    watch.Stop();

    Console.WriteLine($"{ i } => {watch.ElapsedMilliseconds} milliseconds");
}

0 => 80 milliseconds
1 => 6 milliseconds
2 => 1 milliseconds
3 => 3 milliseconds
4 => 3 milliseconds
5 => 3 milliseconds
6 => 3 milliseconds
7 => 3 milliseconds
8 => 3 milliseconds
9 => 3 milliseconds
10 => 9 milliseconds
11 => 4 milliseconds
12 => 3 milliseconds
13 => 3 milliseconds
14 => 3 milliseconds
15 => 3 milliseconds
16 => 3 milliseconds
17 => 3 milliseconds
18 => 1 milliseconds
19 => 744 milliseconds
20 => 1017 milliseconds
21 => 1 milliseconds
22 => 1011 milliseconds
23 => 1006 milliseconds
24 => 510 milliseconds
25 => 1021 milliseconds
26 => 1011 milliseconds
27 => 1018 milliseconds
28 => 508 milliseconds
29 => 515 milliseconds
30 => 1024 milliseconds
31 => 510 milliseconds
32 => 1013 milliseconds
33 => 1025 milliseconds
34 => 1025 milliseconds
35 => 1014 milliseconds

**Comment by michaelklishin** [comment]
We cannot tell without a traffic capture or profiler information. IIRC there was a reason in 6.x that's been resolved months ago.

In general, this client does not optimize for such use cases. In fact, [high connection churn](https://rabbitmq.com/connections.html#high-connection-churn) and short-lived connections are considered **really bad practices**.

**Comment by lukebakken** [comment]
Start with a larger thread pool size in your application:

https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/main/projects/TestApplications/CreateChannel/Program.cs#L20

In a production application you would not be creating so many connections at once.

**Reply by davebally** [reply]
I am suffering from the same sort of issue here..  i *have to* create 120+ connections ( multiple vhosts 1 per customer) on application startup.  This is a multi node system and to tell our clients we are processing correctly 'alives' are sent on mq as well as the product information.  If i didnt create them all at once then the system would be seen as down.  I've looked at the source here and wondering why threads are used and not Tasks / async awaits.

**Reply by lukebakken** [reply]
Historical baggage. This will change in the future but for now you'll have to increase the size of the thread pool.

**Reply by davebally** [reply]
Thanks,  is there a recommend number of threads be required MQ connection ?

**Reply by lukebakken** [reply]
Every connection requires a thread, so start with the number of connections you expect.

**Comment by lukebakken** [comment]
@wj8400684 @davebally please take the time to test the next major release of this library, which does not have this issue:

https://www.nuget.org/packages/RabbitMQ.Client/7.0.0-alpha.6
